### PR TITLE
Cmd changes

### PR DESCRIPTION
**Pull request by pdevine**
This changes the API so that it can use `POST /api/generate` to pre-load a model if the prompt is empty. It also changes the REPL so that it can pre-load the model by making a call to generate automatically, and includes a change for using placeholder text.

### CODE REVIEW

**Comment by jmorganca** [comment]
I'm getting some output on load:

```
./ollama run orca-mini
 Sure, I'd be happy to help! Can you please provide more information on what specific tasks or services you need assistance with?

>>> Send a message (/? for help)
```

**Comment by jmorganca** [comment]
Woops, I had an old server.

**Review by jmorganca — APPROVED** [review]

File: cmd/cmd.go [hunk 1]
```diff
@@ -33,6 +33,17 @@ import (
 	"github.com/jmorganca/ollama/version"
 )
 
+type Painter struct{}
+
+func (p Painter) Paint(line []rune, l int) []rune {
+	termType := os.Getenv("TERM")
+	if termType == "xterm-256color" && len(line) == 0 {
+		prompt := "Send a message (/? for help)"
+		return []rune(fmt.Sprintf("\033[38;5;245m%s\033[%dD\033[0m", prompt, len(prompt)))
```
Comment (BruceMacD) [inline cmd/cmd.go:42]: what do these magical formatting numbers do?

Comment (pdevine) [reply cmd/cmd.go:42]: Just to break it down:

`\033[38;5;245m` - turns the text colour to grey
`%s` - print out the "Send a message"
`\033[%dD` - move the cursor back %d chars
`\033[0m` - reset the color to default

the grey is neutral, so should work in either dark or light themed terminals.

**Pull request merged**